crates/primitives-traits/src/transaction/mod.rs

```diff
@@ -1,28 +1,65 @@
-//! Transaction abstraction
+use core::{fmt::Debug, hash::Hash};
 
-pub mod signed;
-
-use alloc::fmt;
+use alloy_primitives::{TxKind, B256};
 
+use alloy_rlp::Encodable;
 use reth_codecs::Compact;
 use serde::{Deserialize, Serialize};
 
-/// Helper trait that unifies all behaviour required by transaction to support full node operations.
-pub trait FullTransaction: Transaction + Compact {}
-
-impl<T> FullTransaction for T where T: Transaction + Compact {}
+pub mod signed;
 
-/// Abstraction of a transaction.
-pub trait Transaction:
-    alloy_consensus::Transaction
+#[allow(dead_code)]
+/// Inner trait for a raw transaction.
+pub trait Transaction<T>:
```

Comment:
```suggestion
pub trait Transaction:
```
we don't this to have a generic, if anything, give it an associated type